### ISSUE TITLE
Backport.System.Threading.Lock build warnings

### ISSUE THREAD

**Issue by rockfordlhotka** [MarkCiliaVincenti/Backport.System.Threading.Lock#4]
Did something change to break backward compat with this library? All of a sudden the CSLA build is full of warnings.

```
Severity	Code	Description	Project	File	Line	Suppression State	Details
Warning (active)	CS0436	The type 'LockFactory' in 'S:\src\rdl\csla\Source\Csla\obj\Debug\net48\Backport.System.Threading.Lock.SourceGenerator\Backport.System.Threading.Lock.SourceGenerator.Generator\Backport.System.Threading.Lock.SourceGenerator.LockFactory.cs' conflicts with the imported type 'LockFactory' in 'Backport.System.Threading.Lock, Version=3.0.3.0, Culture=neutral, PublicKeyToken=b4f1bfae1afe4011'. Using the type defined in 'S:\src\rdl\csla\Source\Csla\obj\Debug\net48\Backport.System.Threading.Lock.SourceGenerator\Backport.System.Threading.Lock.SourceGenerator.Generator\Backport.System.Threading.Lock.SourceGenerator.LockFactory.cs'.	Csla (net48)	S:\src\rdl\csla\Source\Csla\obj\Debug\net48\Csla.GlobalUsings.g.cs	9		
```

There are others, but this one is pretty common.

**Comment by rockfordlhotka** [comment]
Here's a PR with the issues: https://github.com/MarimerLLC/csla/pull/4362

**Comment by MarkCiliaVincenti** [comment]
I'm trying to fix this issue as also mentioned in https://github.com/MarimerLLC/csla/issues/4370 and https://github.com/Aiko-IT-Systems/DisCatSharp/issues/595.

The issue is that the code from the source generator conflicts with the code from the standard library.
@CptWesley do you have any ideas on how we can fix this please? That is how it it should not generate source unless used as a source generator.

**Comment by CptWesley** [comment]
Copy from DM:

> I have noticed some odd behaviour at times when using source generators. If you build from visual studio the source generator it is executed with .NET Framework, while it uses .NET (Core) when executed from the command line. Also the used environments and settings might be different depending on the build type... Usually this doesn't really matter much, but perhaps the logic that you wrote to conditionally include the source code is behaving differently depending on the build environment? It's a bit of a shot in the dark though.

**Comment by CptWesley** [comment]
I haven't really looked into the code you wrote that determines whether to act as source generator and whether to act as regular library. I can maybe take a look next week, could you point me where in the code this currently happens?

**Comment by MarkCiliaVincenti** [comment]
> I haven't really looked into the code you wrote that determines whether to act as source generator and whether to act as regular library. I can maybe take a look next week, could you point me where in the code this currently happens?

Cheers. It's in the csproj of the main project.

**Comment by Lulalaby** [comment]
*subscribing myself*

**Comment by CptWesley** [comment]
@MarkCiliaVincenti PR is up

**Comment by MarkCiliaVincenti** [comment]
This is fixed by v3.1.0
